packages/samples/headless-react/src/components/recommendation-list/recommendation-list.class.tsx

```diff
@@ -0,0 +1,78 @@
+import {Component} from 'react';
+import {ResultLink} from '../result-list/result-link';
+import {
+  buildRecommendationList,
+  RecommendationList as HeadlessRecommendationList,
+  RecommendationListState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+export interface RecommendationListProps {
+  id?: string;
+}
+
+export class RecommendationList extends Component<RecommendationListProps> {
+  private controller: HeadlessRecommendationList;
+  public state: RecommendationListState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: RecommendationListProps) {
+    super(props);
+
+    this.controller = buildRecommendationList(
+      engine,
+      props.id ? {options: {id: props.id}} : {}
```

Comment:
Do you need the check? Or would passing an undefined id work as well?

Reply:
Unfortunately, passing `undefined` gave an error that said the option must be defined.